mlir/lib/Dialect/Tosa/IR/TosaCanonicalizations.cpp

```diff
@@ -874,7 +874,8 @@ OpFoldResult ReverseOp::fold(FoldAdaptor adaptor) {
     return operandAttr;
 
   // If the dim-length is 1, tosa.reverse is a no-op.
-  if (operandTy.hasRank() && operandTy.getDimSize(axis) == 1)
+  if (operandTy.hasRank() && operandTy.getRank() != 0 &&
```

Comment:
Should rank-0 tensors get folded as well?

Reply:
Give it a go. Would be great to write a small canonicalization test as well in `canonicalize.mlir` in TOSA tests for these.